BlueprintUI/Sources/Element/ComposedElement.swift

```diff
@@ -0,0 +1,22 @@
+import CoreGraphics
+
+/// An element built out of other elements.
+///
+/// Similar in purpose to `ProxyElement`, but the contents may change depending on the `Environment`.
+///
+/// - seealso: ProxyElement
+public protocol ComposedElement: Element {
+
+    /// Return the contents of this element in the given environment.
+    func elementRepresentation(in environment: Environment) -> Element
+}
+
+extension ComposedElement {
```

Comment:
I know it'd be a painful refactor, but maybe fine once: Rather than a new element, should we refactor existing base types to always take in the environment?

Reply:
(Eg, in 6 months from now; more people will be using Blueprint – do we want them to have to deal with the separate types? Does it let us simplify anything in here to do this annoying refactor now?)

Reply:
If you don't need the environment, ProxyElement is potentially much more efficient. I don't think we should get rid of it until we can guarantee that ComposedElement is safe as a universal replacement. I would love to only have one eventually though.